server/service/base_client.go

```diff
@@ -24,6 +25,13 @@ type baseClient struct {
 	http               httpClient
 	urlPrefix          string
 	insecureSkipVerify bool
+	// serverCapabilities is a list of capabilities that the server supports.
+	// This map is updated on each response we receive from the server.
+	serverCapabilities map[fleet.Capability]struct{}
+	// clientCapabilities is a list of capabilities that the client supports.
+	// This list is given when the client is instantiated and shouldn't be
+	// modified afterwards.
+	clientCapabilities []fleet.Capability
```

Comment:
Why is one a list and the other a map?

Reply:
Nit: I'd would make this map to cover the case that the user tries to add duplicated capabilities.

Reply:
just because a map seemed overkill for the case, but I see your points, I will update.